File: src/main/cpp/plazma/kernel/lib/text/printer_helper.cpp
#include <string>

#include "printer_helper.h"

#include "plazma/kernel/lib/str/strlib.h"
#include "plazma/kernel/lib/sys/syslib.h"

namespace text {

    static const char* EVENT_PREFIX   = "=> ";
    static const char* EVENT_ITEM_PREFIX   = "   ";

    bool eq(const char* str1, const char* str2) {
        return strlib::equals(str1, str2);
    }

    void println(const std::string &message) {
        syslib::println(message);
    }

    void println(const ext::ustring &message) {
        syslib::println(message);
    }

    void println(const std::string &message1, const std::string &message2) {
        syslib::print(message1);
        syslib::println(message2);
    }

    void println(const std::string &message1, const std::string &message2, const std::string &message3) {
        syslib::print(message1);
        syslib::print(message2);
        syslib::println(message3);
    }

    void println(const std::string &message1, const std::string &message2, const std::string &message3, const std::string &message4) {
        syslib::print(message1);
        syslib::print(message2);
        syslib::print(message3);
        syslib::println(message4);
    }

    ////

    void println(bool flag, const std::string &message) {
        if (!flag) {
            return;
        }
        println(message);
    }

    void println(bool flag, const std::string &message1, const std::string &message2) {
        if (!flag) {
            return;
        }
        println(message1, message2);
    }

    void println(bool flag, const std::string &message1, const std::string &message2, const std::string &message3) {
        if (!flag) {
            return;
        }
        println(message1, message2, message3);
    }

    ////

    void printEvent(bool flag, const std::string &message) {
        if (!flag) {
            return;
        }
        syslib::print(EVENT_PREFIX);
        syslib::print(message);
        syslib::println();
    }

    void printEvent(bool flag, const std::string &message1, const std::string &message2) {
        if (!flag) {
            return;
        }
        syslib::print(EVENT_PREFIX);
        syslib::print(message1);
        syslib::print(message2);
        syslib::println();
    }

    void printEvent2(bool flag, const std::string &message1, const ext::ustring &message2) {
        if (!flag) {
            return;
        }
        syslib::print(EVENT_PREFIX);
        syslib::print(message1);
        syslib::print(message2);
        syslib::println();
    }

    void printEventItem(bool flag, const std::string &message) {
        if (!flag) {
            return;
        }
        println(EVENT_ITEM_PREFIX + message);
    }

    void printEventItem(bool flag, const std::string &message1, const std::string &message2) {
        if (!flag) {
            return;
        }
        println(EVENT_ITEM_PREFIX + message1, message2);
    }

    void printEventItem(bool flag, const std::string &message1, const std::string &message2, const std::string &message3, const std::string &message4) {
        if (!flag) {
            return;
        }
        println(EVENT_ITEM_PREFIX + message1, message2, message3, message4);
    }

    ////

    void info(const std::string &message) {
        syslib::info(message);
    }

    ////

    void warn(const std::string &message) {
        syslib::warn(message);
    }

    ////

    void error(const std::string &message) {
        syslib::error(message);
    }

    void error(const std::string &message1, const std::string &message2) {
        syslib::error(message1 + message2);
    }

    void error(const std::string &message1, const std::string &message2, const std::string &message3) {
        syslib::error(message1 + message2 + message3);
    }

    ////

    std::string toSafeString(char* array) {
        if (array == nullptr) {
            return "";
        }
        return array;
    }


    
}

File: src/main/java/plazma/lib/text/PrinterHelper.java
package plazma.lib.text;

import plazma.lib.sys.SysLib;

public class PrinterHelper {
    
    public static final String EVENT_PREFIX   = "=> ";
    public static final String EVENT_ITEM_PREFIX   = "   ";
    
    ////
    
    public static void println(String message) {
        SysLib.println(message);
    }
    
//    public static void println(String message1, String message2) {
//        println(message1 + message2);
//    }
//
//    public static void println(String message1, String message2, String message3) {
//        println(message1 + message2 + message3);
//    }
//
//    public static void println(String message1, String message2, String message3, String message4) {
//        println(message1 + message2 + message3 + message4);
//    }

    public static void println(String... message) {
        SysLib.println(message);
    }

    public static void println(boolean flag, String message1, String message2) {
        if (!flag) {
            return;
        }
        println(message1, message2);
    }

    public static void println(boolean flag, String message) {
        if (!flag) {
            return;
        }
        println(message);
    }
    
    ////

    public static void printEvent(boolean flag, String message) {
        if (!flag) {
            return;
        }
        SysLib.print(EVENT_PREFIX);
        SysLib.print(message);
        SysLib.println();
    }

    public static void printEvent(boolean flag, String message1, String message2) {
        if (!flag) {
            return;
        }
        SysLib.print(EVENT_PREFIX);
        SysLib.print(message1);
        SysLib.print(message2);
        SysLib.println();
    }

    public static void printEventItem(boolean flag, String message) {
        if (!flag) {
            return;
        }
        println(EVENT_ITEM_PREFIX, message);
    }

    public static void printEventItem(boolean flag, String message1, String message2) {
        if (!flag) {
            return;
        }
        println(EVENT_ITEM_PREFIX, message1, message2);
    }

    public static void printEventItem(boolean flag, String message1, String message2, String message3, String message4) {
        if (!flag) {
            return;
        }
        println(EVENT_ITEM_PREFIX, message1, message2, message3, message4);
    }
    
    ////

    public static void info(String message) {
        SysLib.info(message);
    }

    public static void warn(String message) {
        SysLib.warn(message);
    }

    public static void error(String message) {
        SysLib.error(message);
    }
    
    public static void error(String message1, String message2) {
        SysLib.error(message1 + message2);
    }

    public static void error(String message1, String message2, String message3) {
        SysLib.error(message1 + message2 + message3);
    }
    
    ////

    public static String toSafeString(String str) {
        return str == null ? "" : str;        
    }


}
